### PR TITLE
[mono] make sure that MonoBundledResource.id is valid

### PR DESCRIPTION
**Pull request by pavelsavara**
I saw ASAN on browser/wasm to complain about NULL derefence in `key_from_id` while loading time zone bundle. 
But I was not able to reproduce it without ASAN.

This is just trying to add defensive code.

### CODE REVIEW

**Comment by mdh1418** [comment]
cc: @kg

What is ASAN?

**Review by mdh1418 — APPROVED** [review]
Looks good to me!

**Comment by kg** [comment]
asan is llvm address sanitizer, it catches various bad pointers. it can also catch leaks.

a related feature is ubsan, which detects undefined behavior

**Review by kg — APPROVED** [review]
sorry :(

id really shouldn't be null here...

File: src/mono/mono/metadata/bundled-resources.c [hunk 1]
```diff
@@ -78,19 +78,24 @@ bundled_resources_is_known_assembly_extension (const char *ext)
...
+	}
 	if (!buffer) {
 		// Add space for .dll and null terminator
 		buffer_len = (guint)(id_length + 6);
 		buffer = g_malloc (buffer_len);
 	}
 	buffer[0] = 0;
 
-	if (extension_offset && bundled_resources_is_known_assembly_extension (extension)) {
+	if (extension_offset != -1 && bundled_resources_is_known_assembly_extension (extension)) {
```
Comment (kg) [inline src/mono/mono/metadata/bundled-resources.c:98]: i prefer 'extension_offset >= 0'

**Pull request merged**